guide/src/generate_text_using_the_vertex_ai_gemini_api.md

```diff
@@ -0,0 +1,127 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Generate text using the Vertex AI Gemini API
+
+In this guide, you send a text prompt request, and then a multimodal prompt and
+image request to the Vertex AI Gemini API and view the responses.
+
+## Prerequisites
+
+Completing this guide requires you to set up a Google Cloud project and enable
+the Vertex AI API. We recommend you use the [Vertex AI Quickstart] to complete
+these steps.
+
+## Adding the Vertex AI client library as a dependency
+
+The Vertex AI client library includes many features. Compiling all of them is
+relatively slow. To speed up compilation, you can just enable the features you
+need:
+
+```toml
+{{#include ../samples/Cargo.toml:aiplatform}}
+```
+
+## Send a prompt to the Vertex AI Gemini API
+
+First, initialize the client using the default settings:
+
+```rust,ignore,noplayground
+{{#include ../samples/src/gemini.rs:text-prompt-client}}
+```
+
+We need to build the model name. For simplicity, this example receives the
+project id as an argument and uses a fixed location (`global`) and model id
+(`gemini-2.0-flash-001`).
+
+If you want to run this function in your own code, use the project id (without
+any `projects/` prefix) of the project you selected while going through the
+prerequisites.
+
+```rust,ignore,noplayground
+{{#include ../samples/src/gemini.rs:text-prompt-model}}
+```
+
+With the client initialized we can send the request:
+
+```rust,ignore,noplayground
+{{#include ../samples/src/gemini.rs:text-prompt-request}}
+```
+
+And then print the response. We use the `:#?` format specifier to prettify the
+nested response objects.
+
+```rust,ignore,noplayground
+{{#include ../samples/src/gemini.rs:text-prompt-response}}
+```
+
+See [below](#text-prompt-complete-code)
+for the complete code.
+
+## Send a prompt and an image to the Vertex AI Gemini API
+
+As in the previous example, initialize the client using the default settings:
+
+```rust,ignore,noplayground
+{{#include ../samples/src/gemini.rs:prompt-and-image-client}}
+```
+
+And then build the model name.
+
+```rust,ignore,noplayground
+{{#include ../samples/src/gemini.rs:prompt-and-image-model}}
+```
+
+The new request includes an image part:
+
+```rust,ignore,noplayground
+{{#include ../samples/src/gemini.rs:prompt-and-image-image-part}}
+```
+
+and the prompt part:
+
+```rust,ignore,noplayground
+{{#include ../samples/src/gemini.rs:prompt-and-image-prompt-part}}
+```
+
+The full request is constructed thus:
+
+```rust,ignore,noplayground
+{{#include ../samples/src/gemini.rs:prompt-and-image-request}}
+```
+
+As in the previous example, we print the full response.
+
+```rust,ignore,noplayground
+{{#include ../samples/src/gemini.rs:prompt-and-image-response}}
+```
+
+See [below](#prompt-and-image-complete-code)
+for the complete code.
+
```

Comment:
I think we have been breaking this up with a line break

https://github.com/googleapis/google-cloud-rust/blob/ebbd04e221d90b5871d9a745fa8ac824d3a2dbc0/guide/src/mocking_lros.md?plain=1#L234

Reply:
Done.